source/Halibut/ServiceModel/RequestCancellationTokens.cs

```diff
@@ -11,28 +11,41 @@ public RequestCancellationTokens(CancellationToken connectingCancellationToken,
         {
             ConnectingCancellationToken = connectingCancellationToken;
             InProgressRequestCancellationToken = inProgressRequestCancellationToken;
-        }
-
-        public CancellationToken ConnectingCancellationToken { get; set; }
-        public CancellationToken InProgressRequestCancellationToken { get; set; }
-
-        public CancellationToken LinkedCancellationToken => LazyLinkedCancellationToken.Value;
 
-        Lazy<CancellationToken> LazyLinkedCancellationToken => new (() =>
-        {
             if (ConnectingCancellationToken == CancellationToken.None && InProgressRequestCancellationToken == CancellationToken.None)
             {
-                return CancellationToken.None;
+                LinkedCancellationToken = CancellationToken.None;
+            }
+            else if (InProgressRequestCancellationToken == CancellationToken.None)
+            {
+                LinkedCancellationToken = ConnectingCancellationToken;
+            }
+            else if (ConnectingCancellationToken == CancellationToken.None)
+            {
+                LinkedCancellationToken = InProgressRequestCancellationToken;
+            }
+            else if (ConnectingCancellationToken == InProgressRequestCancellationToken)
+            {
+                LinkedCancellationToken = ConnectingCancellationToken;
             }
+            else
+            {
+                linkedCancellationTokenSource = CancellationTokenSource.CreateLinkedTokenSource(ConnectingCancellationToken, InProgressRequestCancellationToken);
 
-            linkedCancellationTokenSource = CancellationTokenSource.CreateLinkedTokenSource(ConnectingCancellationToken, InProgressRequestCancellationToken);
+                LinkedCancellationToken = linkedCancellationTokenSource.Token;
+            }
+        }
+
+        public CancellationToken ConnectingCancellationToken { get; set; }
```

Comment:
A bit late to the party, but why are these two property setters public? Why not just read only properties?